website/project/views/drafts.py

```diff
@@ -0,0 +1,221 @@
+from flask import request, redirect
+import httplib as http
+from dateutil.parser import parse as parse_date
+import functools
+
+from modularodm import Q
+from modularodm.exceptions import ValidationValueError
+
+from framework import status
+from framework.mongo import database
+from framework.mongo.utils import get_or_http_error, autoload
+from framework.exceptions import HTTPError
+from framework.status import push_status_message
+
+from website.exceptions import NodeStateError
+from website.util import sanitize
+from website.util.permissions import ADMIN
+from website.project.decorators import (
+    must_be_valid_project,
+    must_have_permission,
+    http_error_if_disk_saving_mode
+)
+from website import language
+from website.project import utils as project_utils
+from website.project.model import MetaSchema, DraftRegistration
+from website.project.metadata.utils import serialize_meta_schema, serialize_draft_registration
+from website.project.utils import serialize_node
+
+get_schema_or_fail = lambda query: get_or_http_error(MetaSchema, query)
+autoload_draft = functools.partial(autoload, DraftRegistration, 'draft_id', 'draft')
+
+# TODO: uncomment to allow submitting drafts for review
+# @must_have_permission(ADMIN)
+# @must_be_valid_project
+# def submit_draft_for_review(auth, node, draft_pk, *args, **kwargs):
+#     user = auth.user
+#
+#     draft = get_draft_or_fail(draft_pk)
+#     draft.is_pending_review = True
+#     draft.save()
+#
+#     REVIEW_EMAIL = Mail(tpl_prefix='prereg_review', subject='New Prereg Prize registration ready for review')
+#     send_mail(draft.initiator.email, REVIEW_EMAIL, user=user, src=node)
+#
+#     ret = project_utils.serialize_node(node, auth)
+#     ret['success'] = True
+#     return ret
+
+@autoload_draft
+@must_have_permission(ADMIN)
+@must_be_valid_project
+def draft_before_register_page(auth, node, draft, *args, **kwargs):
+    ret = serialize_node(node, auth, primary=True)
+
+    ret['draft'] = serialize_draft_registration(draft, auth)
+    return ret
+
+@autoload_draft
+@must_have_permission(ADMIN)
+@must_be_valid_project
+@http_error_if_disk_saving_mode
+def register_draft_registration(auth, node, draft, *args, **kwargs):
+
+    # TODO(hrybacki): Move to framework.utils.rapply once @sam's PR#4027 is merged.
+    from api.base.serializers import _rapply
+    data = _rapply(request.get_json(), sanitize.strip_html)
```

Comment:
Sanitization is handy, but not necessary if the user inputs are never stored in the database as is. (and if side effects would mess up the few things that actually are done with the data)

The only case things we do with this input in this view are an exact-comparison, and construction of a datetime object. Because neither provides a vector to get unsanitized data into the database- and because we don't want bleach transforming something used in an exact comparison- I'm going to do the unprecedented and suggest taking `strip_html` back *out* of this particular function.